test/ecdsatest.c

```diff
@@ -57,343 +49,269 @@ static int restore_rand(void)
     return 1;
 }
 
-static int fbytes_counter = 0, use_fake = 0;
-static const char *numbers[8] = {
-    "651056770906015076056810763456358567190100156695615665659",
-    "6140507067065001063065065565667405560006161556565665656654",
-    "8763001015071075675010661307616710783570106710677817767166"
-        "71676178726717",
-    "7000000175690566466555057817571571075705015757757057795755"
-        "55657156756655",
-    "1275552191113212300012030439187146164646146646466749494799",
-    "1542725565216523985789236956265265265235675811949404040041",
-    "1456427555219115346513212300075341203043918714616464614664"
-        "64667494947990",
-    "1712787255652165239672857892369562652652652356758119494040"
-        "40041670216363"
-};
+static int use_fake = 0;
+static const char *numbers[2];
 
 static int fbytes(unsigned char *buf, int num)
 {
     int ret = 0;
+    static int fbytes_counter = 0;
     BIGNUM *tmp = NULL;
 
     if (use_fake == 0)
         return old_rand->bytes(buf, num);
 
     use_fake = 0;
 
-    if (fbytes_counter >= 8)
+    if (fbytes_counter >= 2)
         return 0;
     if (!TEST_ptr(tmp = BN_new()))
         return 0;
-    if (!TEST_true(BN_dec2bn(&tmp, numbers[fbytes_counter]))) {
+    if (!TEST_true(BN_hex2bn(&tmp, numbers[fbytes_counter]))) {
         BN_free(tmp);
         return 0;
     }
-    fbytes_counter++;
-    if (TEST_int_eq(BN_num_bytes(tmp), num)
-            && TEST_true(BN_bn2bin(tmp, buf)))
+    fbytes_counter ^= 1;
+    /* tmp might need leading zeros so pad it out */
+    if (TEST_int_le(BN_num_bytes(tmp), num)
+            && TEST_true(BN_bn2binpad(tmp, buf, num)))
         ret = 1;
     BN_free(tmp);
     return ret;
 }
 
-/* some tests from the X9.62 draft */
-static int x9_62_test_internal(int nid, const char *r_in, const char *s_in)
+/*-
+ * This function hijacks the RNG to feed it the chosen ECDSA key and nonce.
+ * The ECDSA KATs are from:
+ * - the X9.62 draft (4)
+ * - NIST CAVP (720)
+ *
+ * It uses the low-level ECDSA_sign_setup instead of EVP to control the RNG.
+ * NB: This is not how applications should use ECDSA; this is only for testing.
+ *
+ * Tests the library can successfully:
+ * - generate public keys that matches those KATs
+ * - create ECDSA signatures that match those KATs
+ * - accept those signatures as valid
+ */
+static int x9_62_tests(int n)
 {
-    int ret = 0;
-    const char message[] = "abc";
-    unsigned char digest[SHA_DIGEST_LENGTH];
+    int nid, md_nid, ret = 0;
+    const char *r_in, *s_in, *tbs = NULL;
+    unsigned char *pbuf, *qbuf, *message = NULL;
```

Comment:
As catched by the CI and mentioned by @kroeckx , `pbuf` and `qbuf` should also initialized to `NULL`

Reply:
What's the magic `config` option to make my own build barf on this? To test before I fixup.

Reply:
It's compiling through the clang in Xcode 9.3

Reply:
and with `--strict-warnings`

Reply:
Thanks! fixup in adb59a11447dbcb320a35ab395e0c863756afa5c